### PR TITLE
Update codeowners and CI config for AOT

### PR DESCRIPTION
**Pull request by srnagar**
This PR updates the codeowners for `sdk/aot/` directory, includes perf and samples packages as additional modules and creates javadoc jar with README.

### CODE REVIEW

**Review by JimSuplizio — APPROVED** [review]
@srnagar this may or may not actually appease the MSDocs. Either way, having the readme instead of a pure empty javadoc is the right way to go.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -6,7 +6,7 @@
 ################
 
 # Git Hub integration and bot rules
-/.github/                                       @AlexGhiondea @jsquire
+/.github/                                           @AlexGhiondea @jsquire
```
Comment (alzimmermsft) [inline .github/CODEOWNERS:9]: @AlexGhiondea, @jsquire, do you want to scope this to the `fabricbot.json` file so you aren't pinged for review when CODEOWNERs change?

**Review by alzimmermsft — APPROVED** [review]

**Comment by jsquire** [comment]
@alzimmermsft:  Please do not.   We need visibility on CODEOWNERS changes in order to keep the bot rules up to date with "Service Attention" contacts or when paths for PR labels change.

File: .github/CODEOWNERS (hunk 1 above)
Comment (jsquire) [reply .github/CODEOWNERS:9]: @alzimmermsft:  Please do not change this.   We need visibility on CODEOWNERS changes in order to keep the bot rules up to date with "Service Attention" contacts or when paths for PR labels change.

Comment (alzimmermsft) [reply .github/CODEOWNERS:9]: Got it, didn't realize it was tied closely to CODEOWNERS

Comment (jsquire) [reply .github/CODEOWNERS:9]: Unfortunately, it is - and its currently a manual sync due to bot limitations.

**Pull request merged**